kratos/solving_strategies/convergencecriterias/residual_criteria.h

```diff
@@ -221,7 +222,9 @@ class ResidualCriteria
         BaseType::InitializeSolutionStep(rModelPart, rDofSet, rA, rDx, rb);
 
         // Filling mActiveDofs when MPC exist
-        ConstraintUtilities::ComputeActiveDofs(rModelPart, mActiveDofs, rDofSet);
+        if (rModelPart.NumberOfMasterSlaveConstraints() > 0) {
```

Comment:
I see, but it is checked inside the utility...

Reply:
https://github.com/KratosMultiphysics/Kratos/blob/6c1072408aa41ba67b70ba9caf92f982f51f8980/kratos/utilities/constraint_utilities.cpp#L50

Reply:
[this line](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/utilities/constraint_utilities.cpp#L45) is the problem, this is not protected

In any case this is only a hotfix, will need proper revision when the constraints are parallelized

Reply:
What do you mean is not protected?, each component of the vector is independent

Reply:
you cannot access the vector based in the EquationIds when running in MPI. This is because they in most cases don't start from 0 (only on one rank)
e.g. when running with 2 processes and eq-sys-size 100, then rank 0 has ids 0-50 and rank 1 has 51-100
now if you use the ids to access the vector (which has size 50) you access out of bounds which caused the problems for @sunethwarna (and any other other MPI-case but none else noticed)

Basically we need a global-to-local Id map to solve this properly, but this will be a longer discussion